tests/e2e/cht-form/mock-config.js

```diff
@@ -0,0 +1,62 @@
+const express = require('express');
+const path = require('path');
+const fs = require('fs/promises');
+const generateXformService = require('../../../api/src/services/generate-xform');
+const genericForm = require('@page-objects/default/enketo/generic-form.wdio.page');
+
+let server;
+const mockApp = express();
+mockApp.get('/', (req, res) => res.sendFile(path.join(__dirname, 'index.html')));
+mockApp.use(express.static(path.join(__dirname, '../../../build/cht-form')));
+
+const getFormPath = (config, formType, formName) => {
+  if (formType === 'test') {
+    return path.join(__dirname, config, 'forms', `${formName}.xml`);
+  }
+  return path.join(__dirname, '../../../config', config, 'forms', formType, `${formName}.xml`);
+};
+
+const generateFormData = async (formPath) => {
+  const formXml = await fs.readFile(formPath, 'utf-8');
+  // This is essentially the same code used by the test-harness to generate the form html and model.
+  // If this code changes, the test-harness will need to be updated as well.
+  const { form: formHtml, model: formModel } = await generateXformService.generate(formXml);
+  return { formHtml, formModel, formXml };
+};
+
+const startMockApp = async (config, formType, formName) => {
```

Comment:
:warning: please don't implement this change, I'm just thinking out loud for potential improvements :warning: 

I would love if we could write e2e test files in typescript. Ideally these parameters would have dynamic types that map to the actual configs and forms available
```ts
mockConfig.startMockApp('default', 'app', 'pregnancy');
//                                        ^ autocompletes to 'death_report' | 'delivery' | 'pregnancy'... and so on when the first arg is 'default'
//                      ^ autocompletes to 'default' | 'standard' | 'covid-19' | 'demo'
```

Reply:
This is something that Josh and I had talked about on different occasions, it is a great idea, and it is something that we should do also for the e2e tests in `cht-core`